framework/graphics/dx12_util.cpp

```diff
@@ -703,6 +703,26 @@ void TrackAdapters(HRESULT result, void** ppFactory, graphics::dx12::ActiveAdapt
     }
 }
 
+void RemoveDeactiveAdapters(graphics::dx12::ActiveAdapterMap& adapters)
+{
+    std::vector<int64_t> deactive_adapters;
+    for (auto itr = adapters.begin(); itr != adapters.end(); ++itr)
```

Comment:
I suggest just `for(const auto& adapter: adapters)` here, then you can use `adapter.second` and `adapter.first` in the loop body.